worlds/oot_soh/location_access/overworld/gerudo_fortress.py

```diff
@@ -4,6 +4,315 @@
+    ])
+
+    # GF HBA Range
+    # Locations
+    add_locations(Regions.GF_HBA_RANGE, world, [
+        (Locations.GF_HBA_1000_POINTS,
+         lambda bundle: is_adult(bundle) and has_item(Items.CHILD_WALLET, bundle) and has_item(
+             Items.GERUDO_MEMBERSHIP_CARD, bundle) and can_use(Items.EPONA, bundle) and can_use(Items.FAIRY_BOW,
+                                                                                                bundle) and at_day(
+             bundle)),
+        (Locations.GF_HBA_1500_POINTS,
+         lambda bundle: is_adult(bundle) and has_item(Items.CHILD_WALLET, bundle) and has_item(
+             Items.GERUDO_MEMBERSHIP_CARD, bundle) and can_use(Items.EPONA, bundle) and can_use(Items.FAIRY_BOW,
+                                                                                                bundle) and at_day(
+             bundle)),
+        (Locations.GF_GS_ARCHERY_RANGE,
+         lambda bundle: is_adult(bundle) and can_get_enemy_drop(bundle, Enemies.GOLD_SKULLTULA,
+                                                                EnemyDistance.BOOMERANG) and can_get_nighttime_gs(
+             bundle)),
+        (Locations.GF_HBA_RANGE_CRATE_1, lambda bundle: can_break_crates(bundle)),
+        (Locations.GF_HBA_RANGE_CRATE_2, lambda bundle: can_break_crates(bundle)),
+        (Locations.GF_HBA_RANGE_CRATE_3, lambda bundle: can_break_crates(bundle)),
+        (Locations.GF_HBA_RANGE_CRATE_4, lambda bundle: can_break_crates(bundle)),
+        (Locations.GF_HBA_RANGE_CRATE_5, lambda bundle: can_break_crates(bundle)),
+        (Locations.GF_HBA_RANGE_CRATE_6, lambda bundle: can_break_crates(bundle)),
+        (Locations.GF_HBA_RANGE_CRATE_7, lambda bundle: can_break_crates(bundle)),
+        (Locations.GF_HBA_CANOPY_EAST_CRATE, lambda bundle: can_break_crates(bundle)),
+        (Locations.GF_HBA_CANOPY_WEST_CRATE, lambda bundle: can_break_crates(bundle)),
+        (Locations.GF_NORTH_TARGET_EAST_CRATE, lambda bundle: can_break_crates(bundle)),
+        (Locations.GF_NORTH_TARGET_WEST_CRATE, lambda bundle: is_adult(bundle) or (
+                    blast_or_smash(bundle) or hookshot_or_boomerang(bundle) or can_use(Items.HOVER_BOOTS, bundle))),
+        (Locations.GF_NORTH_TARGET_CHILD_CRATE, lambda bundle: is_child(bundle) and blast_or_smash(bundle)),
+        (Locations.GF_SOUTH_TARGET_EAST_CRATE, lambda bundle: can_break_crates(bundle)),
+        (Locations.GF_SOUTH_TARGET_WEST_CRATE, lambda bundle: can_break_crates(bundle)),
+    ])
+    # Connections
+    connect_regions(Regions.GF_HBA_RANGE, world, [
+        (Regions.GF_OUTSIDE_GTG, lambda bundle: is_child(bundle) or has_item(Items.GERUDO_MEMBERSHIP_CARD, bundle)),
+    ])
+
+    # GF Outside Gate
+    # Events
+    add_events(Regions.GF_OUTSIDE_GATE, world, [
+        (EventLocations.GF_GATE, LocalEvents.GF_GATE_OPEN,
+         lambda bundle: is_adult(bundle) and has_item(Items.GERUDO_MEMBERSHIP_CARD, bundle)),
+    ])
```

Comment:
EventLocations need to have unique names, so this'll need to be something like `GF_GATE_OUTSIDE` (or if you can think of anything better lol).